gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/check_snowy_block.mcfunction

```diff
@@ -0,0 +1,15 @@
+# clear existing tags, check if the player is on a snowy block and wearing skis, add the skiing tag back if yes, and add ski_off_snow tag if not on snowy block
+# @s = player wearing skis
+# at @s
+# run from skis_equipped
+
+# Remove tags unless the player is in the air
+execute unless block ~ ~-0.1 ~ minecraft:air run tag @s remove gm4_mountaineering_skiing
+execute unless block ~ ~-0.1 ~ minecraft:air run tag @s remove gm4_mountaineering_skiing_off_snow
```

Comment:
this can call the `gm4_mountaineering:ski_effects/ski_off_snow_slowness/remove_off_snow_slowness`, and where you add the tag in this function you can call `gm4_mountaineering:ski_effects/ski_off_snow_slowness/add_off_snow_slowness`, which would remove the need for those function tags in `skis_equipped`. Make sure you move the tag checks here so it doesn't run when it doesn't need to